### PR TITLE
[Function](ELT)Add elt function for doris

### LINKED ISSUE #11990: [Enhancement] Add elt function for doris.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Returns the string at specified index. Returns NULL if there is no string at specified index.

Syntax:
`VARCHAR elt (INT,VARCHAR,...)`

Example:
mysql> select elt(1, 'aaa', 'bbb');
+----------------------+
 elt(1, 'aaa', 'bbb')         
+----------------------+
 aaa                                 
+----------------------+
mysql> select elt(2, 'aaa', 'bbb');
+-----------------------+
 elt(2, 'aaa', 'bbb')          
+-----------------------+
 bbb                                 
+-----------------------+
mysql> select elt(0, 'aaa', 'bbb');
+-----------------------+
 elt(0, 'aaa', 'bbb')          
+-----------------------+
 NULL                              
+-----------------------+
mysql> select elt(3, 'aaa', 'bbb');
+-----------------------+
 elt(3, 'aaa', 'bbb')          
+-----------------------+
 NULL                               
+-----------------------+

### Solution

Add elt function for doris.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yongjinhou**
# Proposed changes

Issue Number: close #11990

## Problem summary

Add elt function for doris.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments
 
Add elt function for doris.

### CODE REVIEW

**Review by yangzhg — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
Is there a better name for this function?  ELT is a proper noun for extract load and transform.

**Comment by yongjinhou** [comment]
> 
This is a representation borrowed from mysql.

File: be/src/vec/functions/function_string.h [hunk 1]
```diff
@@ -494,6 +494,51 @@ class FunctionStringConcat : public IFunction {
...
+class FunctionStringElt : public IFunction {
+public:
+    static constexpr auto name = "elt";
+    static FunctionPtr create() { return std::make_shared<FunctionStringElt>(); }
+    String get_name() const override { return name; }
+    size_t get_number_of_arguments() const override { return 0; }
+    bool is_variadic() const override { return true; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeString>();
```
Comment (cambyzju) [inline be/src/vec/functions/function_string.h:506]: need check arguments size >= 2?

File: gensrc/script/doris_builtins_functions.py [hunk 2]
```diff
@@ -2012,6 +2012,12 @@
     [['concat'], 'VARCHAR', ['VARCHAR', '...'],
             '_ZN5doris15StringFunctions6concatEPN9doris_udf15FunctionContextEiPKNS1_9StringValE',
             '', '', 'vec', ''],
+    [['elt'], 'VARCHAR', ['INT', 'VARCHAR', '...'],
+            '_ZN5doris15StringFunctions3eltEPN9doris_udf15FunctionContextERKNS1_6IntValEiPKNS1_9StringValE',
+            '', '', 'vec', ''],  
+    [['elt'], 'STRING', ['INT', 'STRING', '...'],
```
Comment (cambyzju) [inline gensrc/script/doris_builtins_functions.py:2018]: elt should always return NULLABLE? not depends on arguments.

**Comment by TaoZex** [comment]
@yangzhg Hello，I would like to ask how to generate the newly added function signature?

**Comment by yongjinhou** [comment]
> @yangzhg Hello，I would like to ask how to generate the newly added function signature by name mangling?
nm doris_be | grep elt | grep FunctionContext
Run the preceding command.
doris_be  is the binary generated by compiling BE
elt is function name

**Comment by TaoZex** [comment]
> > @yangzhg Hello，I would like to ask how to generate the newly added function signature by name mangling?
> > nm doris_be | grep elt | grep FunctionContext
> > Run the preceding command.
> > doris_be  is the binary generated by compiling BE
> > elt is function name

What are the following steps in the development process? 
1. Build a development environment 
3. Write function source code and put it in BE 
4. Compile BE and generate function signature through name mangling 
5. Test the code function (but I don't know how to test it yet) 
6. Submit pr

**Comment by yongjinhou** [comment]
> > > @yangzhg Hello，I would like to ask how to generate the newly added function signature by name mangling?
> > > nm doris_be | grep elt | grep FunctionContext
> > > Run the preceding command.
> > > doris_be  is the binary generated by compiling BE
> > > elt is function name
> 
> What are the following steps in the development process?
> 
> 1. Build a development environment
> 2. Write function source code and put it in BE
> 3. Compile BE and generate function signature through name mangling
> 4. Test the code function (but I don't know how to test it yet)
> 5. Submit pr

you can click "https://www.bilibili.com/video/BV1V44y1Y7Ro/?spm_id_from=333.788&vd_source=897445b3e6d43614801a4aa3f1b3b57c" to learn more and see "Files changed" in this PR.

**Comment by TaoZex** [comment]
> > > > @yangzhg Hello，I would like to ask how to generate the newly added function signature by name mangling?
> > > > nm doris_be | grep elt | grep FunctionContext
> > > > Run the preceding command.
> > > > doris_be  is the binary generated by compiling BE
> > > > elt is function name
> > 
> > 
> > What are the following steps in the development process?
> > 
> > 1. Build a development environment
> > 2. Write function source code and put it in BE
> > 3. Compile BE and generate function signature through name mangling
> > 4. Test the code function (but I don't know how to test it yet)
> > 5. Submit pr
> 
> you can click "https://www.bilibili.com/video/BV1V44y1Y7Ro/?spm_id_from=333.788&vd_source=897445b3e6d43614801a4aa3f1b3b57c" to learn more and see "Files changed" in this PR.

Thanks.

**Pull request merged**